drivers/iio/adc/ad7768-1.c

```
@@ -599,48 +615,57 @@ static int ad7768_probe(struct spi_device *spi)
 	spi_set_drvdata(spi, indio_dev);
 	mutex_init(&st->lock);
 
+	st->spi_engine_supported = spi_engine_offload_supported(spi);
+
 	indio_dev->channels = ad7768_channels;
 	indio_dev->num_channels = ARRAY_SIZE(ad7768_channels);
 	indio_dev->dev.parent = &spi->dev;
 	indio_dev->name = spi_get_device_id(spi)->name;
 	indio_dev->info = &ad7768_info;
-	indio_dev->modes = INDIO_DIRECT_MODE | INDIO_BUFFER_TRIGGERED;
+	spi_buffer_mode = st->spi_engine_supported ? INDIO_BUFFER_HARDWARE :
+						     INDIO_BUFFER_TRIGGERED;
+	indio_dev->modes = INDIO_DIRECT_MODE | spi_buffer_mode;
 
 	ret = ad7768_setup(st);
 	if (ret < 0) {
 		dev_err(&spi->dev, "AD7768 setup failed\n");
 		return ret;
 	}
 
-	st->trig = devm_iio_trigger_alloc(&spi->dev, "%s-dev%d",
-					  indio_dev->name, indio_dev->id);
-	if (!st->trig)
-		return -ENOMEM;
+	if (st->spi_engine_supported){
+		return -ENOSYS;
+	} else {
```

why the `else` [and the indentation of the code below] if 
```
if (st->spi_engine_supported)
    return -ENOSYS;
```

also, checkpatch will complain for ENOSYS;
maybe use  ENOTSUPP

---

curios: should it be:
```
if (!st->spi_engine_supported)
    return - ENOTSUPP;
```
or; do we need to return an error code?
can the driver work with offload and without?
depending on the case, it may be better to just return 0